### ISSUE TITLE
RuntimeError: Assigned array dtype (int16) cannot accommodate number of category labels (32833)

### ISSUE THREAD

**Issue by zaikunzhang** [sophiamyang/condastats#11]
* condastats version: 0.1.5
* Python version: 3.9.7
* Operating System: Ubuntu 20.04

### Description

Running
```
>>> import condastats.cli
>>> condastats.cli.overall('numpy')
```
I received
```
RuntimeError: Assigned array dtype (int16) cannot accommodate number of category labels (32833)
```

### What I Did

```
>>> import condastats.cli
>>> condastats.cli.overall('numpy')

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/zaikunzhang/.local/lib/python3.9/site-packages/condastats/cli.py", line 85, in overall
    df = df.compute()
  File "/home/zaikunzhang/.local/lib/python3.9/site-packages/dask/base.py", line 312, in compute
    (result,) = compute(self, traverse=False, **kwargs)
  File "/home/zaikunzhang/.local/lib/python3.9/site-packages/dask/base.py", line 600, in compute
    results = schedule(dsk, keys, **kwargs)
  File "/home/zaikunzhang/.local/lib/python3.9/site-packages/dask/threaded.py", line 81, in get
    results = get_async(
  File "/home/zaikunzhang/.local/lib/python3.9/site-packages/dask/local.py", line 508, in get_async
    raise_exception(exc, tb)
  File "/home/zaikunzhang/.local/lib/python3.9/site-packages/dask/local.py", line 316, in reraise
    raise exc
  File "/home/zaikunzhang/.local/lib/python3.9/site-packages/dask/local.py", line 221, in execute_task
    result = _execute_task(task, data)
  File "/home/zaikunzhang/.local/lib/python3.9/site-packages/dask/core.py", line 119, in _execute_task
    return func(*(_execute_task(a, cache) for a in args))
  File "/home/zaikunzhang/.local/lib/python3.9/site-packages/dask/optimization.py", line 990, in __call__
    return core.get(self.dsk, self.outkey, dict(zip(self.inkeys, args)))
  File "/home/zaikunzhang/.local/lib/python3.9/site-packages/dask/core.py", line 149, in get
    result = _execute_task(task, cache)
  File "/home/zaikunzhang/.local/lib/python3.9/site-packages/dask/core.py", line 119, in _execute_task
    return func(*(_execute_task(a, cache) for a in args))
  File "/home/zaikunzhang/.local/lib/python3.9/site-packages/dask/core.py", line 119, in <genexpr>
    return func(*(_execute_task(a, cache) for a in args))
  File "/home/zaikunzhang/.local/lib/python3.9/site-packages/dask/core.py", line 113, in _execute_task
    return [_execute_task(a, cache) for a in arg]
  File "/home/zaikunzhang/.local/lib/python3.9/site-packages/dask/core.py", line 113, in <listcomp>
    return [_execute_task(a, cache) for a in arg]
  File "/home/zaikunzhang/.local/lib/python3.9/site-packages/dask/core.py", line 119, in _execute_task
    return func(*(_execute_task(a, cache) for a in args))
  File "/home/zaikunzhang/.local/lib/python3.9/site-packages/dask/dataframe/io/parquet/core.py", line 87, in __call__
    return read_parquet_part(
  File "/home/zaikunzhang/.local/lib/python3.9/site-packages/dask/dataframe/io/parquet/core.py", line 565, in read_parquet_part
    dfs = [
  File "/home/zaikunzhang/.local/lib/python3.9/site-packages/dask/dataframe/io/parquet/core.py", line 566, in <listcomp>
    func(fs, rg, columns.copy(), index, **toolz.merge(kwargs, kw))
  File "/home/zaikunzhang/.local/lib/python3.9/site-packages/dask/dataframe/io/parquet/fastparquet.py", line 1001, in read_partition
    return cls.pf_to_pandas(
  File "/home/zaikunzhang/.local/lib/python3.9/site-packages/dask/dataframe/io/parquet/fastparquet.py", line 1093, in pf_to_pandas
    pf.read_row_group_file(
  File "/home/zaikunzhang/.local/lib/python3.9/site-packages/fastparquet/api.py", line 361, in read_row_group_file
    core.read_row_group(
  File "/home/zaikunzhang/.local/lib/python3.9/site-packages/fastparquet/core.py", line 608, in read_row_group
    read_row_group_arrays(file, rg, columns, categories, schema_helper,
  File "/home/zaikunzhang/.local/lib/python3.9/site-packages/fastparquet/core.py", line 580, in read_row_group_arrays
    read_col(column, schema_helper, file, use_cat=name+'-catdef' in out,
  File "/home/zaikunzhang/.local/lib/python3.9/site-packages/fastparquet/core.py", line 476, in read_col
    raise RuntimeError('Assigned array dtype (%s) cannot accommodate '
RuntimeError: Assigned array dtype (int16) cannot accommodate number of category labels (32833)
```

Thank you very much!

**Comment by sophiamyang** [comment]
Hi, thanks for the bug report. This is due to [a fastparquet issue](https://github.com/dask/fastparquet/issues/789). I plan to change my code to use pyarrow to fix it when I get time. In the meantime, you can calculate stats from the package public data directly: 

```
df = dd.read_parquet(
        "s3://anaconda-package-data/conda/monthly/*/*.parquet",
        storage_options={"anon": True},
        engine="pyarrow"
    )
```

**Comment by zaikunzhang** [comment]
Thank you for the explanation!

**Comment by sharadraju** [comment]
condastats version: 0.1.5
Python version: 3.8.8
Operating System: Windows 10
I am also getting the same error when I run _condastats_ overall pandas on my Windows 10 Command Line:
`(base) C:\Users\SCRAJU>condastats overall pandas
Traceback (most recent call last):
  File "C:\Users\SCRAJU\anaconda3\Scripts\condastats-script.py", line 9, in <module>
    sys.exit(main())
  File "C:\Users\SCRAJU\anaconda3\lib\site-packages\condastats\cli.py", line 382, in main
    overall(
  File "C:\Users\SCRAJU\anaconda3\lib\site-packages\condastats\cli.py", line 85, in overall
    df = df.compute()
  File "C:\Users\SCRAJU\anaconda3\lib\site-packages\dask\base.py", line 284, in compute
    (result,) = compute(self, traverse=False, **kwargs)
  File "C:\Users\SCRAJU\anaconda3\lib\site-packages\dask\base.py", line 566, in compute
    results = schedule(dsk, keys, **kwargs)
  File "C:\Users\SCRAJU\anaconda3\lib\site-packages\dask\threaded.py", line 79, in get
    results = get_async(
  File "C:\Users\SCRAJU\anaconda3\lib\site-packages\dask\local.py", line 514, in get_async
    raise_exception(exc, tb)
  File "C:\Users\SCRAJU\anaconda3\lib\site-packages\dask\local.py", line 325, in reraise
    raise exc
  File "C:\Users\SCRAJU\anaconda3\lib\site-packages\dask\local.py", line 223, in execute_task
    result = _execute_task(task, data)
  File "C:\Users\SCRAJU\anaconda3\lib\site-packages\dask\core.py", line 121, in _execute_task
    return func(*(_execute_task(a, cache) for a in args))
  File "C:\Users\SCRAJU\anaconda3\lib\site-packages\dask\core.py", line 121, in <genexpr>
    return func(*(_execute_task(a, cache) for a in args))
  File "C:\Users\SCRAJU\anaconda3\lib\site-packages\dask\core.py", line 121, in _execute_task
    return func(*(_execute_task(a, cache) for a in args))
  File "C:\Users\SCRAJU\anaconda3\lib\site-packages\dask\dataframe\io\parquet\core.py", line 381, in read_parquet_part
    dfs = [
  File "C:\Users\SCRAJU\anaconda3\lib\site-packages\dask\dataframe\io\parquet\core.py", line 382, in <listcomp>
    func(fs, rg, columns.copy(), index, **toolz.merge(kwargs, kw))
  File "C:\Users\SCRAJU\anaconda3\lib\site-packages\dask\dataframe\io\parquet\fastparquet.py", line 788, in read_partition
    parquet_file.read_row_group_file(
  File "C:\Users\SCRAJU\anaconda3\lib\site-packages\fastparquet\api.py", line 210, in read_row_group_file
    core.read_row_group_file(
  File "C:\Users\SCRAJU\anaconda3\lib\site-packages\fastparquet\core.py", line 303, in read_row_group_file
    return read_row_group(f, rg, columns, categories, schema_helper, cats,
  File "C:\Users\SCRAJU\anaconda3\lib\site-packages\fastparquet\core.py", line 354, in read_row_group
    read_row_group_arrays(file, rg, columns, categories, schema_helper,
  File "C:\Users\SCRAJU\anaconda3\lib\site-packages\fastparquet\core.py", line 329, in read_row_group_arrays
    read_col(column, schema_helper, file, use_cat=name+'-catdef' in out,
  File "C:\Users\SCRAJU\anaconda3\lib\site-packages\fastparquet\core.py", line 209, in read_col
    raise RuntimeError('Assigned array dtype (%s) cannot accommodate '
RuntimeError: Assigned array dtype (int16) cannot accommodate number of category labels (32833)`

**Comment by phwuil** [comment]
Hello, 

Are there any news or workarounds for this issue ? IS this move from `fastparquet` to `pyarrow`  still being considered ?

Thanks a lot !

**Comment by sophiamyang** [comment]
Sorry it took so long. I was secretly hoping fastparquet could fix itself. I released a new version 0.2.1 of condastats, which uses pyarrow in the backend. Please let me know if there are still issues :) Thank you!
